admin/vite.config.ts

```diff
@@ -0,0 +1,53 @@
+import react from "@vitejs/plugin-react-swc";
+import { resolve } from "path";
+import { defineConfig, loadEnv } from "vite";
+import { createHtmlPlugin } from "vite-plugin-html";
+
+import { environment as envVarsToLoad } from "./src/environment";
+
+// https://vitejs.dev/config/
+export default defineConfig(({ mode }) => {
+    const envVars = loadEnv(mode, process.cwd(), "");
+    return {
+        plugins: [
+            react(),
+            createHtmlPlugin({
+                minify: true,
+                entry: resolve(__dirname, "src/loader.ts"),
+                inject: {
+                    data: {
+                        environmentValues: envVarsToLoad
+                            .map(
+                                (envVarKey) => `window.EXTERNAL__${envVarKey}__ = '${mode === "production" ? `$${envVarKey}` : envVars[envVarKey]}';`,
+                            )
+                            .join("\n"),
+                    },
+                },
+            }),
+        ],
+        server: {
+            port: Number(envVars.ADMIN_PORT || 8002),
```

Comment:
```suggestion
            port: Number(envVars.ADMIN_PORT || 8000),
```
We're using 8000 in all our projects, I wouldn't change that. But why do we even need the fallback?

Reply:
Could be useful if 8000 is already used by another process. This was directly taken from webpack.config.ts.

Reply:
I vote for removing.

Reply:
ok, I removed it. Now it's just using the environment entry.